### PR TITLE
Dnadales/787 conformance tests with top level chain predicate failures

### PR DESCRIPTION
**Pull request by dnadales**

### CODE REVIEW

File: cardano-ledger/test/Test/Cardano/Chain/Block/Model.hs [hunk 1]
```diff
@@ -157,6 +186,35 @@ elaborateAndUpdate config (cvs, abstractToConcreteIdMaps) (ast, ab) =
...
+                           config
+                           dCert
+                           chainValidationState
+                           abstractBlock
+      where
+        dCert = rcDCert
+                  (abstractBlock ^. Abstract.bHeader . Abstract.bhIssuer)
+                  stableAfter
+                  abstractState
+          where
```
Comment (nc6) [inline cardano-ledger/test/Test/Cardano/Chain/Block/Model.hs:214]: Bikeshedding: not bother with the nested `where` clauses?

File: cardano-ledger/test/Test/Cardano/Chain/Block/Model.hs [hunk 2]
```diff
@@ -412,3 +471,119 @@ applyTrace tr =
...
+  setMaxHeaderSize <$> Gen.integral (Range.constant 0 maxSize)
+  where
+    setMaxHeaderSize newSize = state { cvsUpdateState = newUpdateState }
+      where
+        newUpdateState =
+          (cvsUpdateState state) { adoptedProtocolParameters = newParameters }
+          where
+            newParameters =
+              (adoptedProtocolParameters (cvsUpdateState state)) { ppMaxHeaderSize = newSize }
+              -- And this is why lenses where invented...
```
Comment (nc6) [inline cardano-ledger/test/Test/Cardano/Chain/Block/Model.hs]: :-D

Comment (nc6) [inline cardano-ledger/test/Test/Cardano/Chain/Block/Model.hs]:
```suggestion
              -- And this is why lenses were invented...
```

**Review by nc6 — APPROVED** [review]
LGTM

File: cardano-ledger/test/Test/Cardano/Chain/Block/Model.hs (hunk 2 above)
Comment (dnadales) [reply cardano-ledger/test/Test/Cardano/Chain/Block/Model.hs]: I guess I wanted to add an extra `where` :rofl: 

Thanks!

File: cardano-ledger/test/Test/Cardano/Chain/Block/Model.hs (hunk 1 above)
Comment (dnadales) [reply cardano-ledger/test/Test/Cardano/Chain/Block/Model.hs:214]: I kind of like them, but I feel like a Taliban (hello NSA) when using them :)

**Pull request merged**